File: Android/app/src/main/java/com/caffeine/swingit/RainGenerator.kt
package com.caffeine.swingit

import com.caffeine.swingit.Graphics.GInterval
import com.caffeine.swingit.Graphics.GPoint
import com.caffeine.swingit.Graphics.IGUpdatable

class RainGenerator(val scene: GameScene) : IGUpdatable
{
    private var lastGenerationTime = 0L
    private val intervalGeneration = GInterval(0f, scene.size.width)


    init {

    }


    override fun update(currentTime: Long)
    {
        if(currentTime - lastGenerationTime > 0L) {
            lastGenerationTime = currentTime
            scene.addChild(Rain(
                    scene.RAIN_SPEED,
                    scene.RAIN_SIZE,
                    GPoint(intervalGeneration.random(), -scene.RAIN_SIZE.height/2)
            ))

            scene.addChild(Rain(
                    scene.RAIN_SPEED * .8f,
                    scene.RAIN_SIZE_SMALL,
                    GPoint(intervalGeneration.random(), -scene.RAIN_SIZE_SMALL.height/2)
            ))
        }
    }
}

File: iOS/Swing-It/Swing-It/RainGenerator.swift
//
//  RainGenerator.swift
//  Swing-It
//
//  Created by Olivier Picard on 01/10/2018.
//  Copyright © 2018 Caffeine. All rights reserved.
//

import Foundation
import SpriteKit

class RainGenerator : IUpdatable
{
    var scene: GameScene
    var lastGenerationTime: TimeInterval = 0
    
    
    init(scene: GameScene)
    {
        self.scene = scene
    }
    
    
    func update(_ currentTime: TimeInterval) {
        lastGenerationTime = currentTime
        scene.addChild(Rain(scene: scene,
                             size: scene.RAIN_SIZE,
                             position: CGPoint(x: CGFloat.random(in: 0...scene.size.width),
                                               y: scene.size.height + scene.RAIN_SIZE.height),
                             speed: scene.RAIN_SPEED,
                             opacity: 1))
        
        scene.addChild(Rain(scene: scene,
                            size: scene.RAIN_SIZE_SMALL,
                            position: CGPoint(x: CGFloat.random(in: 0...scene.size.width),
                                              y: scene.size.height + scene.RAIN_SIZE_SMALL.height),
                            speed: scene.RAIN_SPEED * 0.8,
                            opacity: 0.7))
    }
}
